mllib/src/main/scala/org/apache/spark/ml/tuning/TrainValidationSplit.scala

```diff
@@ -161,11 +169,26 @@ class TrainValidationSplit @Since("1.5.0") (@Since("1.5.0") override val uid: St
     }
 
     // Wait for all metrics to be calculated
-    val metrics = metricFutures.map(ThreadUtils.awaitResult(_, Duration.Inf))
-
-    // Unpersist training & validation set once all metrics have been produced
-    trainingDataset.unpersist()
-    validationDataset.unpersist()
+    val metrics = try {
+      metricFutures.map(ThreadUtils.awaitResult(_, Duration.Inf))
+    }
+    catch {
+      case e: Throwable =>
+        subTaskFailed = true
+        throw e
+    }
+    finally {
+      if (subTaskFailed) {
+        Thread.sleep(1000)
```

Comment:
Would you mind elabourating why we should sleep here? I should avoid relying on sleep in the main codes or test codes whenever possible.